packages/core/src/html/linkProcessor.js

```diff
@@ -87,18 +87,34 @@ function convertMdAndMbdExtToHtmlExt(node) {
     if (hasNoConvert) {
       return;
     }
+
     const { href } = node.attribs;
-    const hasMdExtension = href.slice(-3) === '.md';
-    if (hasMdExtension) {
-      const newHref = `${href.substring(0, href.length - 3)}.html`;
-      node.attribs.href = newHref;
+
+    const hasMdExtension = href.includes('.md');
+    const hasMbdExtension = href.includes('.mbd');
+
+    const isExclusiveOr = (!hasMdExtension && hasMbdExtension) || (hasMdExtension && !hasMbdExtension);
+    if (!isExclusiveOr) {
+      // 1) has both '.md' and '.mbd' extension OR 2) has neither of the extensions
       return;
     }
-    const hasMbdExtension = href.slice(-4) === '.mbd';
-    if (hasMbdExtension) {
-      const newHref = `${href.substring(0, href.length - 4)}.html`;
+
+    const ext = hasMdExtension ? '.md' : '.mbd';
+
+    const extIdx = href.indexOf(ext);
+    const pathWithoutExtension = href.substring(0, extIdx);
+
+    const hasUrlFragment = href.includes(`${ext}#`);
```

Comment:
could use https://nodejs.org/api/url.html to simplify things here

Reply:
Thanks for the tip! Will use the API :-)